pmm/pmm2-release.groovy

```diff
@@ -549,40 +544,42 @@ ENDSSH
                         echo "We will be tagging repos with a tag: $TAG"
 
                         for PACKAGE in "${!repos[@]}"; do
-                            REPO=${repos["$PACKAGE"]}
-                            # Example of an entry in 'copy.list':
-                            # percona-dashboards-2.31.0-19.2209151640.25fba72.el9.x86_64.rpm
-                            SHA=$(grep "$PACKAGE-" copy.list | perl -p -e 's/.*[.]\\d{10}[.]([0-9a-f]{7})[.]el9.*/$1/')
-
-                            if [ -n "$SHA" ] && [ -n "$REPO" ]; then
-                                rm -fr $PACKAGE || true
-                                mkdir $PACKAGE
-                                pushd $PACKAGE >/dev/null
-                                    git clone https://github.com/$REPO ./
-                                    # The default is https, so we want to set it to ssh
-                                    git remote set-url origin git@github.com:$REPO.git
-                                    git checkout $SHA
-                                    echo "SHA: $(git rev-parse HEAD)"
-
-                                    git tag --message="Version $TAG." --sign $TAG
-
-                                    # If the tag already exists, we want to delete it and re-tag this SHA
-                                    if [ $? -eq 128 ]; then
-                                        git tag --delete $TAG
-                                        git push --delete origin $TAG
-                                        git tag --message="Version $TAG." --sign $TAG
-                                    fi
-
-                                    if [ $? -eq 0 ]; then
-                                        git push origin $TAG
-                                    else
-                                        echo "Error: $?"
-                                    fi
-                                popd >/dev/null
-                            else
-                                echo "Warning: the repository $REPO won't get tagged with ${VERSION}"
-                            fi
+                            rm -fr $PACKAGE || true
+                            mkdir -p $PACKAGE
+
+                            pushd $PACKAGE >/dev/null
+                                REPO=${repos["$PACKAGE"]}
+                                git clone https://github.com/$REPO ./
+                                # The default is https, and we want to set it to ssh
+                                git remote set-url origin git@github.com:$REPO.git
+                                
+                                BRANCH="pmm-${VERSION}"
+                                git checkout "$BRANCH"
+                                if [ $? -ne 0 ]; then
+                                  continue
+                                fi
```

Comment:
shouldn't we fail if we are not able to checkout branch?

Reply:
> shouldn't we fail if we are not able to checkout branch?

No, the ticket says this part shouldn't fail the whole pipeline. However, we can do a better job of leaving a trace in the logs.

Reply:
> I also would suggest adding `git clean -fd`

I see no benefit to it - every repo is cloned afresh. So nothing to clean IMHO.